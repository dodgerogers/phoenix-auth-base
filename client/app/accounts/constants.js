import mirrorCreator from 'mirror-creator';

export const actionTypes = mirrorCreator([
  'GET_CURRENT_USER_REQUEST',
  'GET_CURRENT_USER_SUCCESS',
  'GET_CURRENT_USER_FAILURE',
  'GET_CURRENT_USER_PROFILES_REQUEST',
  'GET_CURRENT_USER_PROFILES_SUCCESS',
  'GET_CURRENT_USER_PROFILES_FAILURE',
  'SET_CURRENT_USER_PROFILE',
  'GET_CURRENT_ACCOUNT_REQUEST',
  'REMOVE_CURRENT_ACCOUNT_REQUEST',
]);
