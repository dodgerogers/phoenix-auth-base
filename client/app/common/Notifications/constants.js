import mirrorCreator from 'mirror-creator';

export const actionTypes = mirrorCreator([
  'NOTIFY_SUCCESS',
  'NOTIFY_FAILURE',
  'DESTROY',
]);

export const areaIDs = mirrorCreator([
  'APPLICATION',
  'AUTHENTICATION',
]);
