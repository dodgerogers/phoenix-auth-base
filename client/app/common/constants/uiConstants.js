export const colors = {
  RedRibbon: '#E90F3A',
  Carmine: '#a70122',
  RedViolet: '#ea1aa3',
  HaitiPurple: '#120d36',
  ElectricViolet: '#a923e9',
  WindsorPurple: '#580c79',
  ParisMPurple: '#270864',
  PurpleHeart: '#612cc8',
  JacksonsPurple: '#1f28a5',
  CuriousBlue: '#2a8cbd',
  Turquoise: '#2fd6c2',
  Elm: '#1b9487',
  Shamrock: '#2ed583',
  GenoaGreen: '#1a985a',
  Apple: '#2dd535',
  BilbaoGreen: '#138119',
  AtlantisGreen: '#b4d535',
  Olivetone: '#627516',
  HokeyPokeyGold: '#d6aa32',
  ReefGold: '#9e791b',
  Tango: '#ef7d25',
  HawaiianTan: '#a05315',
  Pomegranate: '#ee3b1f',
  Tabasco: '#a72913',
  Sangria: '#8f030a',
  DullLavender: '#9b89e6',
  Crimson: '#e61740',
};
