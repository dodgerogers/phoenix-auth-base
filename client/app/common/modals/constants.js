import mirrorCreator from 'mirror-creator';

export const actionTypes = mirrorCreator([
  'INITIALIZE_MODAL',
  'SHOW_MODAL',
  'HIDE_MODAL',
]);

export const modalIds = mirrorCreator([
  'LOGIN_MODAL',
  'REGISTRATION_MODAL',
  'CONFIRMATION_MODAL',
  'RESEND_CONFIRMATION_MODAL',
  'FORGOT_PASSWORD_MODAL',
  'RESET_PASSWORD_MODAL',
])
