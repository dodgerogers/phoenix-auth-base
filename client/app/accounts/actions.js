import { actionTypes } from './constants';

export const getCurrentUserRequest = () => ({
  type: actionTypes.GET_CURRENT_USER_REQUEST,
  accessToken,
});

export const getCurrentUserSuccess = user => ({
  type: actionTypes.GET_CURRENT_USER_SUCCESS,
  user,
});

export const getCurrentUserFailure = () => ({
  type: actionTypes.GET_CURRENT_USER_FAILURE,
});

export const getCurrentUserProfilesRequest = () => ({
  type: actionTypes.GET_CURRENT_USER_PROFILES_REQUEST,
  accessToken,
});

export const getCurrentUserProfilesSuccess = profiles => ({
  type: actionTypes.GET_CURRENT_USER_PROFILES_SUCCESS,
  profiles,
});

export const getCurrentUserProfilesFailure = () => ({
  type: actionTypes.GET_CURRENT_USER_PROFILES_FAILURE,
});

export const setCurrentUserProfile = (profileId) => ({
  type: actionTypes.SET_CURRENT_USER_PROFILE,
  profileId,
})
