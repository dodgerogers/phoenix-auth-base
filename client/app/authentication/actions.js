import { actionTypes } from './constants';

export const verifyTokenRequest = accessToken => ({
  type: actionTypes.VERIFY_TOKEN_REQUEST,
  accessToken,
});

export const verifyTokenSuccess = () => ({
  type: actionTypes.VERIFY_TOKEN_SUCCESS,
});

export const verifyTokenFailure = () => ({
  type: actionTypes.VERIFY_TOKEN_FAILURE,
});

export const refreshTokenRequest = () => ({
  type: actionTypes.REFRESH_TOKEN_REQUEST,
});

export const refreshTokenSuccess = () => ({
  type: actionTypes.REFRESH_TOKEN_SUCCESS,
});

export const refreshTokenFailure = () => ({
  type: actionTypes.REFRESH_TOKEN_FAILURE,
});

export const refreshTokenRequestCancelled = () => ({
  type: actionTypes.REFRESH_TOKEN_REQUEST_CANCELLED,
});

export const authenticateWithStoredToken = () => ({
  type: actionTypes.AUTHENTICATE_WITH_STORED_TOKEN_REQUEST,
});

export const authenticateFailure = error => ({
  type: actionTypes.AUTHENTICATE_FAILURE,
  error,
});

export const signOutRequest = () => ({
  type: actionTypes.SIGN_OUT_REQUEST,
});

export const signOutSuccess = () => ({
  type: actionTypes.SIGN_OUT_SUCCESS,
});

export const signOutFailure = () => ({
  type: actionTypes.SIGN_OUT_FAILURE,
});

export const signInSuccess = response => ({
  type: actionTypes.SIGN_IN_SUCCESS,
  accessToken: response.data.accessToken,
});

export const registerSuccess = (_result, _dispatch, { values }) => ({
  type: actionTypes.REGISTER_SUCCESS,
  formValues: values,
});

export const registerFailure = error => ({
  type: actionTypes.REGISTER_FAILURE,
  error,
});

export const confirmationSuccess = response => ({
  type: actionTypes.CONFIRMATION_SUCCESS,
  accessToken: response.data.accessToken,
});

export const confirmationFailure = (error) => ({
  type: actionTypes.CONFIRMATION_FAILURE,
  error,
});

export const resendConfirmationSuccess = (_result, _dispatch, { values }) => ({
  type: actionTypes.RESEND_CONFIRMATION_SUCCESS,
  formValues: values,
});

export const resendConfirmationFailure = (error) => ({
  type: actionTypes.RESEND_CONFIRMATION_FAILURE,
  error,
});

export const passwordResetRequestSuccess = (_result, _dispatch, { values }) => ({
  type: actionTypes.PASSWORD_RESET_REQUEST_SUCCESS,
  formValues: values,
});

export const resetPasswordSuccess = (_result, _dispatch, { values }) => ({
  type: actionTypes.RESET_PASSWORD_SUCCESS,
  formValues: values,
});
