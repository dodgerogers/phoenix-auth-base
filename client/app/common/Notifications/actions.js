import { actionTypes } from './constants';


export const notify = (message, areaID) => ({
  type: actionTypes.NOTIFY,
  id: areaID,
  notification: {
    level: 'success',
    message,
  },
});

export const destroy = (notification, areaID) => ({
  type: actionTypes.DESTROY,
  id: areaID,
  notification,
});
