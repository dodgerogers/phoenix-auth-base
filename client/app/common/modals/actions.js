import { actionTypes } from './constants';

export function showModal(id) {
  return {
    type: actionTypes.SHOW_MODAL,
    data: { id },
  };
}

export function hideModal(id) {
  return {
    type: actionTypes.HIDE_MODAL,
    data: { id },
  };
}
