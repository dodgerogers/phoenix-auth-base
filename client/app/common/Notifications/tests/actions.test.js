import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { fromJS } from 'immutable';
import * as NotificationActions from '../actions';
import { formIDs, actionTypes } from '../constants';

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);

jest.mock('../../../lib/utils/UidGenerator', () => {
  return jest.fn(() => 1);
});

describe('AuthenticationActions', () => {
  const mockAreaID = 'mockAreaID';
  describe('notify', () => {
    let message;
    beforeEach(() => {
      message = 'message';
    });

    it('dispatches with expected actions', () => {
      expect(NotificationActions.notify(mockAreaID, message)).toMatchSnapshot();
    });
  });

  describe('destroy', () => {
    let message, mockNotification;
    beforeEach(() => {
      message = 'message';
      mockNotification = fromJS({ message });
    });

    it('dispatches with expected actions', () => {
      expect(NotificationActions.destroy(mockAreaID, mockNotification)).toMatchSnapshot();
    });
  });
});
