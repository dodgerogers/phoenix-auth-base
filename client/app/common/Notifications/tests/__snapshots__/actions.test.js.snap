// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthenticationActions notifySuccess dispatches with expected actions 1`] = `
Object {
  "id": "mockAreaID",
  "notification": Object {
    "level": "success",
    "message": "message",
  },
  "type": "NOTIFY_SUCCESS",
}
`;
